File: TI1/TI1.2 SOAP_DOTNET_SINBDD_GR09/02.CLICON/CONUNI_SOAP_DOTNET_CLICON_G09/CONUNI_SOAP_DOTNET_CLICON_G09/ec/edu/monster/controlador/Autenticador.cs
using System;
using ec.edu.monster.servicios;
using ec.edu.monster.vista;

namespace ec.edu.monster.controlador
{
    /// <summary>
    /// Clase para manejar la autenticación del sistema.
    /// <para>
    /// Las credenciales NO se almacenan en el cliente. La validación
    /// se delega al servidor a través de la operación SOAP <c>login</c>.
    /// </para>
    /// </summary>
    public class Autenticador
    {
        private const int MAX_INTENTOS = 3;

        private readonly VistaLogin          _vistaLogin;
        private readonly ClienteConversionSOAP _clienteSOAP;

        public Autenticador()
        {
            _vistaLogin  = new VistaLogin();
            _clienteSOAP = new ClienteConversionSOAP();
        }

        /// <summary>
        /// Realiza el proceso de autenticación contra el servidor SOAP.
        /// </summary>
        /// <returns><c>true</c> si la autenticación es exitosa</returns>
        public bool Autenticar()
        {
            _vistaLogin.MostrarBannerLogin();

            for (int intento = 1; intento <= MAX_INTENTOS; intento++)
            {
                _vistaLogin.MostrarIntento(intento, MAX_INTENTOS);

                string usuario   = _vistaLogin.LeerUsuario();
                string contrasena = _vistaLogin.LeerContrasenaOculta();

                if (ValidarCredencialesEnServidor(usuario, contrasena))
                {
                    _vistaLogin.MostrarMensajeExito();
                    return true;
                }
                else
                {
                    _vistaLogin.MostrarMensajeError(intento, MAX_INTENTOS);
                }
            }

            _vistaLogin.MostrarMensajeBloqueo();
            return false;
        }

        /// <summary>
        /// Delega la validación de credenciales al servidor SOAP.
        /// El cliente nunca conoce las credenciales correctas.
        /// </summary>
        private bool ValidarCredencialesEnServidor(string usuario, string contrasena)
        {
            try
            {
                return _clienteSOAP.Login(usuario, contrasena);
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine($"Error al conectar con el servidor para autenticar: {ex.Message}");
                return false;
            }
        }

        /// <summary>
        /// Cierra el autenticador y libera recursos.
        /// </summary>
        public void Cerrar()
        {
            _vistaLogin?.Cerrar();
            _clienteSOAP?.Cerrar();
        }
    }
}

File: TI1/TI1.1 SOAP_JAVA_SINBDD_GR09/02.CLICON/CONUNI_SOAP_JAVA_CLICON_G09/src/main/java/ec/edu/monster/controlador/Autenticador.java
package ec.edu.monster.controlador;

import ec.edu.monster.servicios.ClienteConversionSOAP;
import ec.edu.monster.vista.VistaLogin;

/**
 * Clase para manejar la autenticación del sistema.
 * <p>
 * Las credenciales NO se almacenan en el cliente. La validación
 * se delega al servidor a través de la operación SOAP {@code login}.
 * </p>
 *
 * @author GR09 - Arquitectura de Software
 */
public class Autenticador {

    private static final int MAX_INTENTOS = 3;

    private final VistaLogin          vistaLogin;
    private final ClienteConversionSOAP clienteSOAP;

    public Autenticador() {
        this.vistaLogin   = new VistaLogin();
        this.clienteSOAP  = new ClienteConversionSOAP();
    }

    /**
     * Realiza el proceso de autenticación contra el servidor.
     *
     * @return {@code true} si la autenticación es exitosa, {@code false} en caso contrario
     */
    public boolean autenticar() {
        vistaLogin.mostrarBannerLogin();

        for (int intento = 1; intento <= MAX_INTENTOS; intento++) {
            vistaLogin.mostrarIntento(intento, MAX_INTENTOS);

            String usuario   = vistaLogin.leerUsuario();
            String contrasena = vistaLogin.leerContrasenaOculta();

            if (validarCredencialesEnServidor(usuario, contrasena)) {
                vistaLogin.mostrarMensajeExito();
                return true;
            } else {
                vistaLogin.mostrarMensajeError(intento, MAX_INTENTOS);
            }
        }

        vistaLogin.mostrarMensajeBloqueo();
        return false;
    }

    /**
     * Delega la validación de credenciales al servidor SOAP.
     * El cliente nunca conoce las credenciales correctas.
     */
    private boolean validarCredencialesEnServidor(String usuario, String contrasena) {
        try {
            return clienteSOAP.login(usuario, contrasena);
        } catch (Exception e) {
            System.err.println("Error al conectar con el servidor para autenticar: " + e.getMessage());
            return false;
        }
    }

    /**
     * Cierra el autenticador y libera recursos.
     */
    public void cerrar() {
        if (vistaLogin != null) {
            vistaLogin.cerrar();
        }
    }
}